river/command/importance.zig

```diff
@@ -0,0 +1,64 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2020 Marten Ringwelski
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+
+const Error = @import("../command.zig").Error;
+const Seat = @import("../Seat.zig");
+const View = @import("../View.zig");
+const ViewStack = @import("../view_stack.zig").ViewStack;
+
+/// Swap the currently focused view with either the view higher or lower in the stack
+pub fn importance(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len < 2) return Error.NotEnoughArguments;
+    if (args.len > 2) return Error.TooManyArguments;
+
+    if (seat.focused != .view)
+        return;
+
+    // Filter out everything that is not part of the current layout
+    if (seat.focused.view.pending.float or seat.focused.view.pending.fullscreen) return;
+
+    const direction = std.meta.stringToEnum(enum { up, down }, args[1]) orelse return Error.UnknownOption;
+
+    const focused_node = @fieldParentPtr(ViewStack(View).Node, "view", seat.focused.view);
+    const output = seat.focused_output;
+    var it = if (direction == .up) ViewStack(View).iter(focused_node, .reverse, output.pending.tags, filter)
+        else if (direction == .down) ViewStack(View).iter(focused_node, .forward, output.pending.tags, filter)
+        else unreachable;
```

Comment:
`direction` can only be `.up` or `.down`, so only a single conditional is required.

Also I think you should use `output.current.tags` here.

Reply:
I took inspiration from `zoom.zig` since both `swap` and `zoom` manipulate the viewstack.
For zoom pending instead of current is used everywhere.

When we use the current state here should we also use it in the filter function?